cpp/example_code/codebuild/tests/codebuild_gtests.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+#include "codebuild_gtests.h"
+#include <fstream>
+#include <aws/core/client/ClientConfiguration.h>
+
+Aws::SDKOptions AwsDocTest::CodeBuild_GTests::s_options;
+std::unique_ptr<Aws::Client::ClientConfiguration> AwsDocTest::CodeBuild_GTests::s_clientConfig;
+static const char ALLOCATION_TAG[] = "CodeBuild_GTEST";
+
+/*
+ * Subclass MockHTTPCLient to respond to credential requests.
+ * Otherwise, the stored responses are returned for credential requests
+ * and not the service API calls.
+ */
+class CustomMockHTTPClient : public MockHttpClient {
+public:
+    explicit CustomMockHTTPClient(
+            const std::shared_ptr<Aws::Http::HttpRequest> &requestTmp) {
```

Comment:
The indentation in this section seems off, but it might just be GH's display.

Reply:
Everything now freshly reformatted.